source/E2E.Tests/Services/BesiegerCamps/BesiegerCampPropertyTests.cs

```diff
@@ -0,0 +1,212 @@
+﻿using E2E.Tests.Environment;
+using E2E.Tests.Environment.Instance;
+using E2E.Tests.Util;
+using HarmonyLib;
+using System.Reflection;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Siege;
+using Xunit.Abstractions;
+using static Common.Extensions.ReflectionExtensions;
+using static TaleWorlds.CampaignSystem.Siege.SiegeEvent;
+
+namespace E2E.Tests.Services.BesiegerCamps
+{
+    public class BesiegerCampPropertyTests : IDisposable
+    {
+        private List<MethodBase> disabledMethods = new();
+
+        private E2ETestEnvironment TestEnvironment { get; }
+
+        private EnvironmentInstance Server => TestEnvironment.Server;
+        private IEnumerable<EnvironmentInstance> Clients => TestEnvironment.Clients;
+
+        private readonly string besiegerCampId;
+        private readonly string siegeEventId;
+        private readonly string siegeEnginesId;
+
+        public void Dispose()
+        {
+            TestEnvironment.Dispose();
+        }
+
+        private void DisableMethods()
+        {
+            disabledMethods = new List<MethodBase> {
+                AccessTools.Method(typeof (MobileParty), nameof(MobileParty.OnPartyJoinedSiegeInternal)),
+                AccessTools.Method(typeof (BesiegerCamp), nameof(BesiegerCamp.SetSiegeCampPartyPosition)),
+                AccessTools.Method(typeof (BesiegerCamp), nameof(BesiegerCamp.InitializeSiegeEventSide))
+            };
+
+            disabledMethods.AddRange(AccessTools.GetDeclaredConstructors(typeof(SiegeEvent)));
+        }
+
+        private T ServerCreateObject<T>(out string objectId)
+        {
+            string? id = null;
+            T? obj = default;
+
+            Server.Call(() =>
+            {
+                obj = GameObjectCreator.CreateInitializedObject<T>();
+                Assert.True(Server.ObjectManager.TryGetId(obj, out id));
+            }, disabledMethods);
+
+            objectId = id!;
+            return obj ?? throw new InvalidOperationException("Failed to create object.");
+        }
+
+        public BesiegerCampPropertyTests(ITestOutputHelper output)
+        {
+            TestEnvironment = new E2ETestEnvironment(output);
+            DisableMethods();
```

Comment:
Move disabledMethods assignment to constructor
```suggestion
            disabledMethods = new List<MethodBase> {
                AccessTools.Method(typeof (MobileParty), nameof(MobileParty.OnPartyJoinedSiegeInternal)),
                AccessTools.Method(typeof (BesiegerCamp), nameof(BesiegerCamp.SetSiegeCampPartyPosition)),
                AccessTools.Method(typeof (BesiegerCamp), nameof(BesiegerCamp.InitializeSiegeEventSide))
            };

            disabledMethods.AddRange(AccessTools.GetDeclaredConstructors(typeof(SiegeEvent)));
```

Reply:
done